src/main/java/io/cdap/plugin/servicenow/sink/ServiceNowSink.java

```diff
@@ -55,17 +67,47 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (collector.getValidationFailures().isEmpty()) {
       conf.validateSchema(stageConfigurer.getInputSchema(), collector);
     }
+    Schema schema = conf.getSchema(collector);
+    if (schema != null) {
+      stageConfigurer.setOutputSchema(schema);
+    } else {
+      stageConfigurer.setOutputSchema(stageConfigurer.getInputSchema());
+    }
 
   }
 
   @Override
-  public void prepareRun(BatchSinkContext batchSinkContext) throws Exception {
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    Schema inputSchema = context.getInputSchema();
+    FailureCollector collector = context.getFailureCollector();
+    conf.validate(collector);
+    collector.getOrThrowException();
+    Configuration hConf = new Configuration();
+    ServiceNowOutputFormat.setOutput(hConf, conf);
+    context.addOutput(Output.of(conf.referenceName, new ServiceNowOutputFormatProvider(hConf)));
 
-  }
+    LineageRecorder lineageRecorder = new LineageRecorder(context, conf.referenceName);
+    lineageRecorder.createExternalDataset(inputSchema);
+    // Record the field level WriteOperation
+    if (inputSchema.getFields() != null && !inputSchema.getFields().isEmpty()) {
+      String operationDescription = String.format("Wrote to Servicenow %s", conf.getTableName());
+      lineageRecorder.recordWrite("Write", operationDescription,
+                                  inputSchema.getFields().stream()
+                                    .map(Schema.Field::getName)
+                                    .collect(Collectors.toList()));
+    }
 
+  }
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    super.initialize(context);
+    this.transformer = new ServiceNowTransformer();
+  }
 
-  public void transform(KeyValue<NullWritable, StructuredRecord> input, Emitter<StructuredRecord> emitter) {
-    emitter.emit(input.getValue());
+  @Override
+  public void transform(StructuredRecord record, Emitter<KeyValue<NullWritable, JsonObject>> emitter) {
+    JsonObject jsonObject = transformer.transform(record);
+    emitter.emit(new KeyValue<>(null, jsonObject));
   }
 
   private void recordLineage(BatchSourceContext context, ServiceNowTableInfo tableInfo) {
```

Comment:
remove this method.

Reply:
Removed. f52f02d4f710d02cead2bed152cda9004a27c735